addons/arsenal/functions/fnc_fillRightPanel.sqf

```diff
@@ -207,146 +199,149 @@ switch (_ctrlIDC) do {
+                ["CfgWeapons", _x, true, true, true] call _fnc_fillRightContainer;
             };
-        } forEach (GVAR(virtualItems) get IDX_VIRT_UNIQUE_UNKNOWN_ITEMS);
+        } forEach (keys (GVAR(virtualItems) get IDX_VIRT_UNIQUE_UNKNOWN_ITEMS));
     };
     // Custom buttons
     default {
         private _items = (GVAR(customRightPanelButtons) param [[RIGHT_PANEL_CUSTOM_BUTTONS] find _ctrlIDC, []]) param [0, []];
 
         if (_items isNotEqualTo []) then {
-            // "Regular" misc. items
-            {
-                if (_x in _items) then {
-                    ["CfgWeapons", _x, true] call _fnc_fill_right_Container;
-                };
-            } forEach (GVAR(virtualItems) get IDX_VIRT_MISC_ITEMS);
-            // Unique items
-            {
-                if (_x in _items) then {
-                    ["CfgWeapons", _x, true, true] call _fnc_fill_right_Container;
-                };
-            } forEach (GVAR(virtualItems) get IDX_VIRT_UNIQUE_MISC_ITEMS);
-            // Unique backpacks
             {
-                if (_x in _items) then {
-                    ["CfgVehicles", _x, false, true] call _fnc_fill_right_Container;
-                };
-            } forEach (GVAR(virtualItems) get IDX_VIRT_UNIQUE_BACKPACKS);
-            // Unique goggles
-            {
-                if (_x in _items) then {
-                    ["CfgGlasses", _x, false, true] call _fnc_fill_right_Container;
-                };
-            } forEach (GVAR(virtualItems) get IDX_VIRT_UNIQUE_GOGGLES);
-            // Unknown items
-            {
-                if (_x in _items) then {
-                    ["CfgWeapons", _x, true, true, true] call _fnc_fill_right_Container;
+
+                switch (true) do {
+                    // "Regular" misc. items
+                    case (_x in (GVAR(virtualItems) get IDX_VIRT_MISC_ITEMS)): {
+                        ["CfgWeapons", _x, true] call _fnc_fillRightContainer;
+                    };
+                    // Unique items
+                    case (_x in (GVAR(virtualItems) get IDX_VIRT_MISC_ITEMS)): {
+                        ["CfgWeapons", _x, true, true] call _fnc_fillRightContainer;
+                    };
+                    // Unique backpacks
+                    case (_x in (GVAR(virtualItems) get IDX_VIRT_MISC_ITEMS)): {
+                        ["CfgVehicles", _x, false, true] call _fnc_fillRightContainer;
+                    };
+                    // Unique goggles
+                    case (_x in (GVAR(virtualItems) get IDX_VIRT_MISC_ITEMS)): {
+                        ["CfgGlasses", _x, false, true] call _fnc_fillRightContainer;
+                    };
+                    // Unknown items
+                    case (_x in (GVAR(virtualItems) get IDX_VIRT_UNIQUE_UNKNOWN_ITEMS)): {
+                        ["CfgWeapons", _x, true, true, true] call _fnc_fillRightContainer;
+                    };
```

Comment:
Either I'm missing something or this didn't come out as expected.

Reply:
> Either I'm missing something or this didn't come out as expected.

?

Reply:
Yeah, the `case` statements are just direct copypastes, which won't do what the comments say.

Reply:
Yea, I screwed up when I copy-pasted. I will address this tonight.